addons/calendar/models/calendar.py

```diff
@@ -323,6 +323,14 @@ def get_next_potential_limit_alarm(self, alarm_type, seconds=None, partner_id=No
                 'rrule': rule
             }
 
+        # determine accessible events
+        events = self.env['calendar.event'].browse(result)
+        event_ids = set(events._filter_access_rules('read').ids)
+        result = {
```

Comment:
This is a bit simpler:

```suggestion
        events = self.env['calendar.event'].browse(result)
        return {
            key: result[key]
            for key in set(events._filter_access_rules('read').ids)
        }
```